src/compose/ComposeBox.js

```diff
@@ -320,15 +315,11 @@ class ComposeBox extends PureComponent<Props, State> {
   };
 
   handleSend = () => {
-    const { dispatch, narrow, caughtUp, _ } = this.props;
+    const { dispatch, _ } = this.props;
     const { message } = this.state;
+    const narrow = this.getDestinationNarrow();
```

Comment:
ComposeBox: Send "stop typing" notification to destination narrow.
    
    This sends the "stop typing" notification to the destination narrow,
    rather than the narrow that the ComposeBox happened to be loaded into.
    
    This isn't totally correct - we may still miss sending some stop typing
    notifications that we should send, if a user makes edits to the message
    and then later edits the topic, but this change is still an improvement.

This change is good, but the commit message should mention that all of this is in the future when we start sending typing notifications for stream messages at all; i.e., this is a latent bug, lying in wait for when we add that feature.

Otherwise it's confusing for someone who knows that we only send typing notifications for PMs, as they try to figure out what scenario the commit message is talking about in which we're sending typing notifications and then the destination narrow changes.

Hmm, in fact: I think even in that future, this won't at least primarily be about the destination narrow *changing*. The typical case where `this.getDestinationNarrow()` is different from `this.props.narrow` -- the only case outside of message-editing scenarios -- is when you're in a stream narrow, in which case the destination narrow is always some topic narrow, so always different from the stream narrow. So the latent bug this is fixing is that when we go to add typing notifications for stream messages, when you're in a stream narrow we'd try to send these stopped-typing notifications to the stream narrow, instead of the specific topic. Probably that would look like not sending them at all -- probably `sendTypingStop` would ignore them, just like it currently does everything but PM narrows.

---

… Hmm and actually, looking at the implementation of `getDestinationNarrow`, a `this.props.narrow` of a stream narrow is the *only* case where it returns something different from the prop. That seems like a bug -- if you're in a topic narrow, start editing a message, and edit the topic input, the right answer should use the value from the topic input.

Does that affect any of the other changes in this branch? I feel like there were some refactorings that relied on `getDestinationNarrow`, and may have turned a latent bug (that `getDestinationNarrow` returns a wrong value) into a live one (by using that value in a case where we didn't before.)

Reply:
I fixed the `getDestinationNarrow` bug in a new commit, but I think that may actually break editing messages. Looking into this more...

Reply:
Oh, I was seeing #4830, which reproduces on master :/

I believe that the code as it is now is no more incorrect than it was previously, but I obviously could be missing something.

Reply:
That fix looks good, thanks! (Modulo the thread at https://github.com/zulip/zulip-mobile/pull/4773#discussion_r665786059 .)